### PR TITLE
Automation cmdlet changes

### PR DESCRIPTION
**Pull request by felixcho-work**
Hi Azure team,

Please review and merge Automation cmdlet changes.  This is for the next release train of 6/1.

Thanks.

Change logs:
- New-AzureRmAutomationRunbook and Import-AzureRmAutomationRunbook supports the new “GraphPowerShell” and “GraphPowerShellWorkflow” (recommended in place of “Graph”) types.
- Start-AzureRmAutomationRunbook supports the -Wait parameter
- New-AzureRmAutomationSchedule and Get-AzureRmAutomationSchedule now support weekly and monthly schedules.
- New-AzureRmAutomationSchedule now takes a Timezone parameter to adjust for daylight savings.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@felixcho-msft__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (felixcho). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

File: src/ResourceManager/Automation/Commands.Automation/Cmdlet/NewAzureAutomationSchedule.cs [hunk 1]
```diff
@@ -95,26 +146,168 @@ protected override void AutomationProcessRecord()
...
+        /// </exception>
+        private Schedule CreateMonthlyScheduleModel()
+        {
+            var dayOfWeek = this.DayOfWeek.HasValue ? this.DayOfWeek.ToString() : null;
+            if ((!string.IsNullOrWhiteSpace(dayOfWeek) && this.DayOfWeekOccurrence == 0) || (string.IsNullOrWhiteSpace(dayOfWeek) && this.DayOfWeekOccurrence != 0))
             {
-                schedule.Frequency = ScheduleFrequency.Day;
-                schedule.Interval = this.DayInterval;
+                throw new Exception("for monthly occurrence, both day of week and occurrence need to be specified");
             }
```
Comment (hovsepm) [inline src/ResourceManager/Automation/Commands.Automation/Cmdlet/NewAzureAutomationSchedule.cs]: move this string to resources.

File: src/ResourceManager/Automation/Commands.Automation.Test/UnitTests/NewAzureAutomationRunbookTest.cs [hunk 2]
```diff
@@ -68,5 +70,47 @@ public void NewAzureAutomationRunbookByNameSuccessfull()
...
+            this.cmdlet.Name = runbookName;
+            this.cmdlet.Type = type;
+            this.cmdlet.SetParameterSet("ByRunbookName");
+            this.cmdlet.ExecuteCmdlet();
+
+            this.mockAutomationClient.Verify(f => f.CreateRunbookByName(resourceGroupName, accountName, runbookName, null, null, RunbookTypeEnum.GraphPowerShell, null, null, false), Times.Once());
+        }
+
+        [TestMethod]
+        public void NewAzureAutomationRunbookTypeValueSameAsSdk()
```
Comment (hovsepm) [inline src/ResourceManager/Automation/Commands.Automation.Test/UnitTests/NewAzureAutomationRunbookTest.cs]: this should be converted to Xunit ASAP. Please provide an ETA.

File: src/ResourceManager/Automation/Commands.Automation/Common/RunbookTypeSdkValue.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            if (Constants.RunbookType.GraphicalPowerShell.Equals(commandValue, StringComparison.OrdinalIgnoreCase))
+            {
+                return RunbookTypeEnum.GraphPowerShell;
+            }
+            else if (Constants.RunbookType.GraphicalPowerShellWorkflow.Equals(commandValue, StringComparison.OrdinalIgnoreCase))
+            {
+                return RunbookTypeEnum.GraphPowerShellWorkflow;
+            }
+            else
+            {
```
Comment (hovsepm) [inline src/ResourceManager/Automation/Commands.Automation/Common/RunbookTypeSdkValue.cs]: no need for else. just return commandValue; here.

**Comment by hovsepm** [comment]
>     {

changes to this cmdlet are not covered with any test. Please add. 

---
Refers to: src/ResourceManager/Automation/Commands.Automation/Cmdlet/NewAzureAutomationSchedule.cs:32 in a4fa899. [](commit_id = a4fa89913016ae94bb12861b10a07c8248467341, deletion_comment = False)

**Comment by hovsepm** [comment]
:clock1:

File: src/ResourceManager/Automation/Commands.Automation.Test/UnitTests/NewAzureAutomationRunbookTest.cs (hunk 2 above)
Comment (markcowl) [inline src/ResourceManager/Automation/Commands.Automation.Test/UnitTests/NewAzureAutomationRunbookTest.cs]: We require tests in Xunit.  MSTest tests are no longer accepted.

File: src/ResourceManager/Automation/Commands.Automation/Cmdlet/NewAzureAutomationSchedule.cs (hunk 1 above)
Comment (felixcho-msft) [inline src/ResourceManager/Automation/Commands.Automation/Cmdlet/NewAzureAutomationSchedule.cs]: Done.

File: src/ResourceManager/Automation/Commands.Automation.Test/UnitTests/NewAzureAutomationRunbookTest.cs (hunk 2 above)
Comment (felixcho-msft) [inline src/ResourceManager/Automation/Commands.Automation.Test/UnitTests/NewAzureAutomationRunbookTest.cs]: Removed the new tests.

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/852/

**Pull request merged**